framework/Models/Code.py

```diff
@@ -590,8 +603,81 @@ def collectOutput(self,finishedJob,output,options=None):
+
 
-    self.addOutputFromExportDictionary(exportDict, output, options, finishedJob.identifier)
+  ###################################################################################
+  ## THIS METHOD NEEDS TO BE REWORKED WHEN THE NEW DATAOBJECT STRUCURE IS IN PLACE ##
+  ###################################################################################
+  def collectOutputFromDataObject(self,exportDict,output):
+    """
+      Method to collect the output from a DataObject (if it is not a dataObject, it just returns a list with one single exportDict)
+      @ In, exportDict, dict, the export dictionary
+                               ({'inputSpaceParams':{var1:value1,var2:value2},
+                                 'outputSpaceParams':{outstreamName1:DataObject1,outstreamName2:DataObject2},
+                                 'metadata':{'metadataName1':value1,'metadataName2':value2}})
+      @ Out, returnList, list, list of export dictionaries
+    """
+    returnList = []
+    if exportDict['outputSpaceParams'].values()[0].__class__.__base__.__name__ != 'Data':
+      returnList.append(exportDict)
+    else:
+      # get the DataObject that is compatible with this output
+      compatibleDataObject = None
+      for dataObj in exportDict['outputSpaceParams'].values():
+        if output.type == dataObj.type:
+          compatibleDataObject = dataObj
+          break
+        if output.type == 'HDF5' and dataObj.type == 'HistorySet':
+          compatibleDataObject = dataObj
+          break
+      if compatibleDataObject is None:
+        # if none found (e.g. => we are filling an HistorySet with a PointSet), we take the first one
+        compatibleDataObject = exportDict['outputSpaceParams'].values()[0]
+      # get the values
+      inputs = compatibleDataObject.getParametersValues('inputs',nodeId = 'RecontructEnding')
+      unstructuredInputs = compatibleDataObject.getParametersValues('unstructuredinputs',nodeId = 'RecontructEnding')
+      outputs = compatibleDataObject.getParametersValues('outputs',nodeId = 'RecontructEnding')
+      metadata = compatibleDataObject.getAllMetadata(nodeId = 'RecontructEnding')
+      inputKeys = inputs.keys() if compatibleDataObject.type == 'PointSet' else inputs.values()[0].keys()
+      # expand inputspace of current RAVEN
+      for i in range(len(compatibleDataObject)):
+        appendDict = {'inputSpaceParams':{},'outputSpaceParams':{},'metadata':{}}
+        appendDict['inputSpaceParams'].update(exportDict['inputSpaceParams'])
+        appendDict['metadata'].update(exportDict['metadata'])
+        if compatibleDataObject.type == 'PointSet':
+          for inKey, value in inputs.items():
+            appendDict['inputSpaceParams'][inKey] = value[i]
+          for inKey, value in unstructuredInputs.items():
+            appendDict['inputSpaceParams'][inKey] = value[i]
+          for outKey, value in outputs.items():
+            appendDict['outputSpaceParams'][outKey] = value[i]
```

Comment:
I may be reading this wrong, but does this always take all the results from the slave raven run, or can it take selective results, as requested by the output data object?

Reply:
For now it takes all results. 
I opened an issue to modify it later. (#363)